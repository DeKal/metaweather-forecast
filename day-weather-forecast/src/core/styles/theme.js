import { createMuiTheme } from '@material-ui/core'

const colors = {
  primary: [
    '#2F1068',
    '#44239F',
    '#6138B0',
    '#8F7ECE',
    '#B6B6E2',
    '#D1D1EE',
    '#EBEBF9',
    '#FAFAFF'
  ],
  orange: [
    '#F28708',
    '#F8B44E',
    '#FAC980',
    '#FCDEB2',
    '#FEF4E5',
    '#F39303',
    '#F85B03'
  ],
  teal: ['#00A5B0', '#6ED9E0', '#B9F0F4'],
  blue: ['#007ABF', '#70C0F9', '#A4D1FC', '#C6DDFF', '#EAF2FF'],
  volcano: ['#F25220', '#F9A084', '#FDD1C4'],
  black: '#000000',
  hardEmphasis: '#212121',
  medEmphasis: '#666666',
  disabledText: '#9E9E9E',
  disabledBg: '#EDEDED',
  white: '#FFFFFF',
  error: '#B00020',
  silver: '#C0C0C0',
  grayscale: '#999999',
  mediumGray: '#C4C4C4'
}
const bodyStyle = {
  fontWeight: 400
}
const headingStyle = {
  fontWeight: 500,
  fontStyle: 'normal'
}

const borders = {
  standard: [
    `1px solid ${colors.primary[4]}`,
    `1px solid ${colors.disabledText}`,
    `1px solid ${colors.black}`,
    `1px solid ${colors.primary[6]}`
  ]
}

const theme = createMuiTheme({
  typography: {
    h1: {
      ...headingStyle,
      fontSize: 76,
      lineHeight: '92px'
    },
    h2: {
      ...headingStyle,
      fontSize: 60,
      lineHeight: '72px'
    },
    h3: {
      ...headingStyle,
      fontSize: 48,
      lineHeight: '60px'
    },
    h4: {
      ...headingStyle,
      fontSize: 32,
      lineHeight: '40px'
    },
    h5: {
      ...headingStyle,
      fontSize: 24,
      lineHeight: '28px',
      letterSpacing: '0.2px'
    },
    h6: {
      ...headingStyle,
      fontSize: 20,
      lineHeight: '24px',
      letterSpacing: '0.2px'
    },
    subtitle1: { ...headingStyle, fontSize: 16, lineHeight: '20px' },
    subtitle2: { ...headingStyle, fontSize: 14, lineHeight: '16px' },
    subtitle3: { ...headingStyle, fontSize: 12, lineHeight: '16px' },
    subtitleCaps: { ...headingStyle, fontSize: 12, lineHeight: '16px' },
    body1: { ...bodyStyle, fontSize: 16, lineHeight: '24px' },
    body2: { ...bodyStyle, fontSize: 14, lineHeight: '16px' },
    body3: { ...bodyStyle, fontSize: 12, lineHeight: '18px' }
  },
  colors,
  borders
})

export default theme
