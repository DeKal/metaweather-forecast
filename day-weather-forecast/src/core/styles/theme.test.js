import theme from 'core/styles/theme'

describe('theme test', () => {
  test('should show correctly theme', () => {
    expect(theme).toMatchSnapshot()
  })
})
