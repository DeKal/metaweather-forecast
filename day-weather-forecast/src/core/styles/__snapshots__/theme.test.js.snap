// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme test should show correctly theme 1`] = `
Object {
  "borders": Object {
    "standard": Array [
      "1px solid #B6B6E2",
      "1px solid #9E9E9E",
      "1px solid #000000",
      "1px solid #EBEBF9",
    ],
  },
  "breakpoints": Object {
    "between": [Function],
    "down": [Function],
    "keys": Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ],
    "only": [Function],
    "up": [Function],
    "values": Object {
      "lg": 1280,
      "md": 960,
      "sm": 600,
      "xl": 1920,
      "xs": 0,
    },
    "width": [Function],
  },
  "colors": Object {
    "black": "#000000",
    "blue": Array [
      "#007ABF",
      "#70C0F9",
      "#A4D1FC",
      "#C6DDFF",
      "#EAF2FF",
    ],
    "disabledBg": "#EDEDED",
    "disabledText": "#9E9E9E",
    "error": "#B00020",
    "grayscale": "#999999",
    "hardEmphasis": "#212121",
    "medEmphasis": "#666666",
    "mediumGray": "#C4C4C4",
    "orange": Array [
      "#F28708",
      "#F8B44E",
      "#FAC980",
      "#FCDEB2",
      "#FEF4E5",
      "#F39303",
      "#F85B03",
    ],
    "primary": Array [
      "#2F1068",
      "#44239F",
      "#6138B0",
      "#8F7ECE",
      "#B6B6E2",
      "#D1D1EE",
      "#EBEBF9",
      "#FAFAFF",
    ],
    "silver": "#C0C0C0",
    "teal": Array [
      "#00A5B0",
      "#6ED9E0",
      "#B9F0F4",
    ],
    "volcano": Array [
      "#F25220",
      "#F9A084",
      "#FDD1C4",
    ],
    "white": "#FFFFFF",
  },
  "direction": "ltr",
  "mixins": Object {
    "gutters": [Function],
    "toolbar": Object {
      "@media (min-width:0px) and (orientation: landscape)": Object {
        "minHeight": 48,
      },
      "@media (min-width:600px)": Object {
        "minHeight": 64,
      },
      "minHeight": 56,
    },
  },
  "overrides": Object {},
  "palette": Object {
    "action": Object {
      "activatedOpacity": 0.12,
      "active": "rgba(0, 0, 0, 0.54)",
      "disabled": "rgba(0, 0, 0, 0.26)",
      "disabledBackground": "rgba(0, 0, 0, 0.12)",
      "disabledOpacity": 0.38,
      "focus": "rgba(0, 0, 0, 0.12)",
      "focusOpacity": 0.12,
      "hover": "rgba(0, 0, 0, 0.04)",
      "hoverOpacity": 0.04,
      "selected": "rgba(0, 0, 0, 0.08)",
      "selectedOpacity": 0.08,
    },
    "augmentColor": [Function],
    "background": Object {
      "default": "#fafafa",
      "paper": "#fff",
    },
    "common": Object {
      "black": "#000",
      "white": "#fff",
    },
    "contrastThreshold": 3,
    "divider": "rgba(0, 0, 0, 0.12)",
    "error": Object {
      "contrastText": "#fff",
      "dark": "#d32f2f",
      "light": "#e57373",
      "main": "#f44336",
    },
    "getContrastText": [Function],
    "grey": Object {
      "100": "#f5f5f5",
      "200": "#eeeeee",
      "300": "#e0e0e0",
      "400": "#bdbdbd",
      "50": "#fafafa",
      "500": "#9e9e9e",
      "600": "#757575",
      "700": "#616161",
      "800": "#424242",
      "900": "#212121",
      "A100": "#d5d5d5",
      "A200": "#aaaaaa",
      "A400": "#303030",
      "A700": "#616161",
    },
    "info": Object {
      "contrastText": "#fff",
      "dark": "#1976d2",
      "light": "#64b5f6",
      "main": "#2196f3",
    },
    "primary": Object {
      "contrastText": "#fff",
      "dark": "#303f9f",
      "light": "#7986cb",
      "main": "#3f51b5",
    },
    "secondary": Object {
      "contrastText": "#fff",
      "dark": "#c51162",
      "light": "#ff4081",
      "main": "#f50057",
    },
    "success": Object {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#388e3c",
      "light": "#81c784",
      "main": "#4caf50",
    },
    "text": Object {
      "disabled": "rgba(0, 0, 0, 0.38)",
      "hint": "rgba(0, 0, 0, 0.38)",
      "primary": "rgba(0, 0, 0, 0.87)",
      "secondary": "rgba(0, 0, 0, 0.54)",
    },
    "tonalOffset": 0.2,
    "type": "light",
    "warning": Object {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#f57c00",
      "light": "#ffb74d",
      "main": "#ff9800",
    },
  },
  "props": Object {},
  "shadows": Array [
    "none",
    "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
    "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
    "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
    "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
    "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
    "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
    "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
    "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
    "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
    "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
    "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
    "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
    "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
    "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
    "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
    "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
    "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
    "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
  ],
  "shape": Object {
    "borderRadius": 4,
  },
  "spacing": [Function],
  "transitions": Object {
    "create": [Function],
    "duration": Object {
      "complex": 375,
      "enteringScreen": 225,
      "leavingScreen": 195,
      "short": 250,
      "shorter": 200,
      "shortest": 150,
      "standard": 300,
    },
    "easing": Object {
      "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
      "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
      "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
      "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
    },
    "getAutoHeightDuration": [Function],
  },
  "typography": Object {
    "body1": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 16,
      "fontWeight": 400,
      "letterSpacing": "0.00938em",
      "lineHeight": "24px",
    },
    "body2": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 14,
      "fontWeight": 400,
      "letterSpacing": "0.01071em",
      "lineHeight": "16px",
    },
    "body3": Object {
      "fontSize": 12,
      "fontWeight": 400,
      "lineHeight": "18px",
    },
    "button": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": "0.875rem",
      "fontWeight": 500,
      "letterSpacing": "0.02857em",
      "lineHeight": 1.75,
      "textTransform": "uppercase",
    },
    "caption": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": "0.75rem",
      "fontWeight": 400,
      "letterSpacing": "0.03333em",
      "lineHeight": 1.66,
    },
    "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
    "fontSize": 14,
    "fontWeightBold": 700,
    "fontWeightLight": 300,
    "fontWeightMedium": 500,
    "fontWeightRegular": 400,
    "h1": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 76,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "-0.01562em",
      "lineHeight": "92px",
    },
    "h2": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 60,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "-0.00833em",
      "lineHeight": "72px",
    },
    "h3": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 48,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "0em",
      "lineHeight": "60px",
    },
    "h4": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 32,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "0.00735em",
      "lineHeight": "40px",
    },
    "h5": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 24,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "0.2px",
      "lineHeight": "28px",
    },
    "h6": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 20,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "0.2px",
      "lineHeight": "24px",
    },
    "htmlFontSize": 16,
    "overline": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": "0.75rem",
      "fontWeight": 400,
      "letterSpacing": "0.08333em",
      "lineHeight": 2.66,
      "textTransform": "uppercase",
    },
    "pxToRem": [Function],
    "round": [Function],
    "subtitle1": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 16,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "0.00938em",
      "lineHeight": "20px",
    },
    "subtitle2": Object {
      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "fontSize": 14,
      "fontStyle": "normal",
      "fontWeight": 500,
      "letterSpacing": "0.00714em",
      "lineHeight": "16px",
    },
    "subtitle3": Object {
      "fontSize": 12,
      "fontStyle": "normal",
      "fontWeight": 500,
      "lineHeight": "16px",
    },
    "subtitleCaps": Object {
      "fontSize": 12,
      "fontStyle": "normal",
      "fontWeight": 500,
      "lineHeight": "16px",
    },
  },
  "zIndex": Object {
    "appBar": 1100,
    "drawer": 1200,
    "mobileStepper": 1000,
    "modal": 1300,
    "snackbar": 1400,
    "speedDial": 1050,
    "tooltip": 1500,
  },
}
`;
